index.js

```diff
@@ -812,6 +861,35 @@ SqlPort.prototype.callSP = function(name, params, flatten, fileName) {
                 }
             }
         });
+        if ($meta.saveAs) {
+            var fileDir = path.dirname($meta.saveAs);
+            return new Promise((resolve, reject) => {
+                fs.access(fileDir, (err) => {
```

Comment:
I think it is better to directly try to create the dir and proceed normally in case of error.code==='EEXIST'
Check notes here: https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback
The notes are about file, but should be the same for folders.

Reply:
Please check it now